vtStorAtaProtocol/Platform/Linux/ProtocolAtaPassThrough.cpp

```diff
@@ -27,164 +28,142 @@ namespace vtStor
 namespace Protocol
 {
 
-    const vtStor::U8 FEATURE_REGISTER_OFFSET = 0;
-    const vtStor::U8 ERROR_REGISTER_OFFSET = 0;
-    const vtStor::U8 COUNT_REGISTER_OFFSET = 1;
-    const vtStor::U8 LBA_LOW_REGISTER_OFFSET = 2;
-    const vtStor::U8 LBA_MID_REGISTER_OFFSET = 3;
-    const vtStor::U8 LBA_HIGH_REGISTER_OFFSET = 4;
-    const vtStor::U8 DEVICE_REGISTER_OFFSET = 5;
-    const vtStor::U8 COMMAND_REGISTER_OFFSET = 6;
-    const vtStor::U8 STATUS_REGISTER_OFFSET = 6;
-    const vtStor::U8 RESERVED_REGISTER_OFFSET = 7;
-    
-   /* eErrorCode cAtaPassThrough::IssueCommand( const DeviceHandle& Handle, std::shared_ptr<const cBufferInterface> Essense, std::shared_ptr<cBufferInterface> DataBuffer )
+eErrorCode cAtaPassThrough::IssueCommand(const DeviceHandle& Handle, std::shared_ptr<const cBufferInterface> Essense, std::shared_ptr<cBufferInterface> DataBuffer)
+{
+    eErrorCode errorCode = eErrorCode::None;
+
+    cBufferFormatter bufferFormatter = cBufferFormatter::Reader(Essense);
+
+    switch (bufferFormatter.GetHeader().Format)
+    {
+    case 1:

```

Comment:
It's same VTStor-Window. About case 1, I asked a.Dien. We have many format of BufferFormater. We will support it in future.